tests/main/apparmor-batch-reload/task.yaml

```diff
@@ -0,0 +1,92 @@
+summary: Ensure that batch-reloading of AppArmor profiles works, including fallback
+
+systems: [ubuntu-1*]
+
+environment:
+    VARIANT/changed: changed
+    VARIANT/unchanged: unchanged
+    AA_CACHE: /var/cache/apparmor/
+    AA_PROFILES: /var/lib/snapd/apparmor/profiles/
+
+prepare: |
+    snap install test-snapd-content-plug test-snapd-tools
+    snap install --edge test-snapd-curl
+    cp /sbin/apparmor_parser /sbin/apparmor_parser.real
+    rm -f /tmp/apparmor-parser-calls.debug
+
+restore: |
+    mv /sbin/apparmor_parser.real /sbin/apparmor_parser
+    rm -f /tmp/apparmor-parser-calls.debug profiles.out
+
+debug: |
+    cat /tmp/apparmor-parser-calls.debug || true
```

Comment:
In case the test fails early before the log gets created; anyway this is now entirely removed per the other comment related to potential symlink attack.